### ISSUE TITLE
Retouch: Do not show the full uncropped image if the retouch module is above the crop module

### ISSUE THREAD

**Issue by mfg92** [darktable-org/darktable#17403]
**Is your feature request related to a problem? Please describe.**
I often find myself in the following scenario:

1. I crop a photo
   ![grafik](https://github.com/user-attachments/assets/eb544c2c-a936-41f6-bcd5-7075903a116b)
2. Then I do some retouching to get rid of stuff that was cut by the crop.
   ![grafik](https://github.com/user-attachments/assets/f1ec4dde-a28b-4250-839c-c5995e8c49a8)
3. I notice that the Retouch module does some smearing of colors that originate outside the crop, in this case green is smearing into the blueish background.
   ![grafik](https://github.com/user-attachments/assets/9f69d8c5-f000-4bc5-b692-b623858b546c)
4. To get rid of this smearing, I move the Retouch module above the Crop module. That way, the Retouch module will not consider the colors outside the crop when it is doing its magic.
   ![grafik](https://github.com/user-attachments/assets/7fe69ab9-76bc-411b-be53-4f66d7648abb)
5. If I now open the Retouch module (e.g. to do more retouching), I see the whole uncropped image again and the smearing comes back. At least until I close the Retouch module again.
   ![grafik](https://github.com/user-attachments/assets/59ab9d17-391a-480a-ab0a-fdbe138a9dd9)

So the major issue here is that I get inconsistent output depending on whether the retouch module is open or closed. 

From my point of view, it makes no sense to "uncrop" when the Retouch module is above the Crop module. The intention behind the uncropping feature was to make it easier to source areas of the image that are outside the crop. But if the Retouch module is above the Crop module, that will not work anyway.

**Describe the solution you'd like**
Showing the full, uncropped image while the Retouch module is open should only be done if the Retouch module is below the Crop module. If the Retouch module is above the Crop module, simply display the cropped image.

**Comment by mfg92** [comment]
Is there anything I can do to put this forward?

**Comment by wpferguson** [comment]
IIRC the full uncropped image is shown so that the uncropped part can serve as a source for some of the retouch operations.

**Comment by mfg92** [comment]
I totally agree on why the change was made to show the full image when working in the retouch module: Have the full image available to source a patch from. Makes perfect sense and was a nice addition.

But I think it was not fully thought through. By default in the pipeline, the cropping happens after the retouch. So the retouch module has access to the whole image.
The problem arises when the user moves the crop module before the retouch module. In this scenario, the retouch module receives the already cropped image as input. Therefore, the retouch module cannot access the image data outside the cropped area. This is perfectly understandable and works as expected. But knowing this, it does not make sense to uncrop the image while retouching in this scenario. Uncropping suggests to the user that they can use data from outside the crop, but that is not possible in this scenario.

Therefore, I suggest to make the uncropping behavior dependent on the relative order between the retouch module and the crop module.

**Comment by jenshannoschwalm** [comment]
@mfg92 thought about this again and i think you are fully right. While implementing the "full-view thing" i just didn't think about modules being in non-default positions.
